flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/ApplicationReconciler.java

```diff
@@ -299,9 +303,92 @@ public boolean reconcileOtherChanges(FlinkResourceContext<FlinkDeployment> ctx)
             return true;
         }
 
+        // check for JobManager exceptions if the REST API server is still up.
+        if (!ReconciliationUtils.isJobInTerminalState(deployment.getStatus())) {
+            observeJobManagerExceptions(ctx, deployment, observeConfig);
```

Comment:
Feels like the exception observation logic should be part of the observer and not the reconciler. The method names and everything points to that. No action is taken here, the job is not modified etc just observed.

Reply:
Moved to ApplicationObserver.

Reply:
Moved to JobStatusObserver